zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/builder/fragment/MultiSelectBuilder.kt

```diff
@@ -0,0 +1,70 @@
+package org.hexworks.zircon.api.builder.fragment
+
+import org.hexworks.cobalt.logging.api.LoggerFactory
+import org.hexworks.zircon.api.builder.Builder
+import org.hexworks.zircon.api.fragment.MultiSelect
+import org.hexworks.zircon.api.fragment.builder.FragmentBuilder
+import org.hexworks.zircon.internal.fragment.impl.DefaultMultiSelect
+
+/**
+ * Builder for a [MultiSelect]
+ */
+class MultiSelectBuilder<T : Any>(val width: Int, val values: List<T>) : FragmentBuilder<MultiSelect<T>, MultiSelectBuilder<T>> {
+
+    private val log = LoggerFactory.getLogger(this::class)
+
+    private var callback: (oldValue: T, newValue: T) -> Unit = { _, _ -> log.warn("No callback defined for a MultiSelect input!")}
+    private var centeredText = true
+    private var toStringMethod: (T) -> String = Any::toString
+    private var clickable: Boolean = false
+
+    /**
+     * The callback to be used when the value changes. It gets the old and the new value as parameters.
+     */
+    fun withCallback(callbackFunction: (oldValue: T, newValue: T) -> Unit) = also {
```

Comment:
This is going to be obsolete soon when `ListProperty` gets implemented in Cobalt. Let's keep it here until then, but keep it in mind for the future.

Reply:
Oh, you are right. The whole idea of this fragment is a prime example for a property to watch / be watched (don't know why that wasn't my first idea, but it would have collided with the old/new value thingy anyways). Looking forward for the ListProperty.

Reply:
Btw you can use `Property<PersistentList<T>>` if you really want to, but it would be a bit awkward. 😆

Reply:
It just occurred to me that you can have a `selectedValue: Property<T>` here and then you don't need the callback at all.

Reply:
And when the label is clicked the property updates with the same value, right? I think that would be much easier. But on the other hand you lose the ability to detect if the value changed or not (which I for one wouldn'nt need).